### DISCUSSION TITLE
Forwarders & Server Failures

### DISCUSSION THREAD

**Discussion by c-michail** [TechnitiumSoftware/DnsServer#1055] [General]
Hello,
  I have setup 4 forwarders WITHOUT Concurrent Forwarding. 
  Why Technitium doesn't try the next forwarder in case of a "Server Failure" ?
  I guess it should respond a Server Failure in case all of the forwarders in the list fail.
  
  Thank you

**Comment by ShreyasZare** [comment]
Thanks for asking. The DNS server returns ServerFailure when the request times out and there is still no answer available. The server does not stop resolving the domain name, but keeps on retrying forwarder and when all attempts fail, then it selects the next forwarder and keeps retrying. Once it gets an answer, it will cache it and any subsequent request for the same domain will use it.

Which is why its recommended to have concurrent forwarding enabled so that the DNS server queries multiple forwarders in parallel and uses the fastest response available.

**Comment by c-michail** [comment]
So i got it right with NO Concurrent Forwarding checked and having 3 Forwarders it will try all 3 forwarders in sequental order and if they fail it will return SRVFail correct ? (eg, Forwarder1 = Google, Forwarder2 = Cloudflare, Forwarder3 = Quad9)

Thank you

**Reply by ShreyasZare** [reply]
That's not how it works. If you send a request to the DNS server and it does not have an answer till 2 sec (Client Timeout) then it will respond with ServerFailure response.

The resolution for the domain name keeps on working with each forwarder with Concurrent Forwarding disabled till it gets an answer. Each forwarder gets retried before trying the next one. If none of them answer then a negative cache entry is made which then causes the DNS server to answer with ServerFailure immediately to all requests till the negative cache expires.

**Comment by c-michail** [comment]
Got it! Thank you
